data-model/src/main/java/io/micronaut/data/model/query/builder/sql/AbstractSqlLikeQueryBuilder2.java

```diff
@@ -1300,11 +1305,11 @@ protected final void appendPropertyRef(AnnotationMetadata annotationMetadata,
      * @param dataType the property data type
      */
     private void appendJsonProjection(StringBuilder sb, DataType dataType) {
-        if (dataType == DataType.STRING) {
-            sb.append(".string()");
-        } else if (dataType.isNumeric() || dataType == DataType.BOOLEAN) {
+        if (dataType.isNumeric() || dataType == DataType.BOOLEAN) {
             // Boolean is represented as number in Oracle (which only supports json view)
-            sb.append(".number()");
+            sb.append(".numberOnly()");
+        } else if (dataType == DataType.STRING) {
+            sb.append(".stringOnly()");
         } else if (dataType == DataType.TIMESTAMP) {
             sb.append(".timestamp()");
         } else if (dataType == DataType.DATE) {
```

Comment:
it seems these can be used in the `SELECT` statement as well and there are others like `dateOnly()`

Should we be optimising them all?